README.md

```diff
@@ -359,6 +359,27 @@ Nullability should always be used in header files, it's preferred in implementat
 
 Generics should be adopted wherever possible in header files - this helps document the code. It's optional in implementation files, though I've found it useful.
 
+When dealing with nested arrays, sets and dictionaries. Generics should look neat and be easily readable, therefore it is not recommended to go beyond displaying generics for the first level when arrays are nested within arrays etc.
+
+**Preferred:**
+
+```objc
+- (void)displayMyArray:(NSArray<NSSet *> *)myArray;
+- (void)displayMyDictionary:(NSDictionary<NSString *, NSArray *> *)myDictionary;
+- (void)displayMySet:(NSSet<NSArray <NSDictionary *> *> *)mySet;
```

Comment:
Damn, after looking at so many generics for ten minutes I couldn't even make out my own code. I'll revise it now :)